### ISSUE TITLE
Some keys don't repeat

### ISSUE THREAD

**Issue by laxcat** [glfw/glfw#1010]
I noticed if you hook up `glfwSetCharCallback` (or `glfwSetCharModsCallback`), then HOLD DOWN a letter key—"Q" for example—the callback only gets called once, and never repeats.

But if you hold down another key—let's say "E"—it DOES repeat, which I would expect to be the correct behavior.

The only pattern I can figure that determines which keys repeat and which don't is that keys that normally have a little diacritics popup (see picture) DO repeat, and keys that don't have that popup, don't repeat.

![screen shot 2017-05-08 at 3 48 35 pm](https://cloud.githubusercontent.com/assets/41416/25826963/2da53b7e-3406-11e7-9cff-2b0e70a8c963.png)

I tested on a couple computers (both on latest version of macOS Sierra) with the same results. I'll do a little more research, like if [disabling that popup](http://osxdaily.com/2011/08/04/enable-key-repeat-mac-os-x-lion/) changes this behavior.

It should also be noted that this only seems to affect `glfwSetCharCallback` and `glfwSetCharModsCallback`. `glfwSetKeyCallback` seems to function normally and repeats for all keys when held.

Test code:
```
// main.cpp
#include <GLFW/glfw3.h>
#include <iostream>

void charCallback(GLFWwindow * window, unsigned int codepoint, int mods) {
    std::cout << "char callback called with codepoint " << codepoint << std::endl;
}

int main(void)
{
    GLFWwindow* window;

    if (!glfwInit())
        return -1;

    window = glfwCreateWindow(640, 480, "Hello World", NULL, NULL);
    if (!window)
    {
        glfwTerminate();
        return -1;
    }

    glfwMakeContextCurrent(window);

    glfwSetCharModsCallback(window, charCallback);

    while (!glfwWindowShouldClose(window))
    {
        glClear(GL_COLOR_BUFFER_BIT);
        glfwSwapBuffers(window);
        glfwPollEvents();
    }

    glfwTerminate();
    return 0;
}
```

Built and run with:
`clang++ main.cpp -I . -L . -lglfw3 -framework Cocoa -framework IOKit -framework CoreVideo -framework OpenGL -o glfwbugtest && ./glfwbugtest`
Where the GLFW includes folder and libglfw3.a are in the same directory as main.cpp.

**Comment by laxcat** [comment]
[Disabling the diacritics popup](http://osxdaily.com/2011/08/04/enable-key-repeat-mac-os-x-lion/) does indeed resolve the issue. Using the above code, all keys now call the callback repeatedly when held.

It's hardly a solution though, as only a fraction of a percent of people will have disabled the popup, and it's not even something you can do through standard settings.

**Comment by elmindreda** [comment]
This is likely dependent on #41.

**Comment by ScrimpyCat** [comment]
You can disable this programmatically in your application by changing your user defaults. e.g. something like this:

```objective-c
#define GLFW_EXPOSE_NATIVE_COCOA
#import <GLFW/glfw3native.h>

void AllowKeyRepeats(void)
{
    @autoreleasepool {
        [[NSUserDefaults standardUserDefaults] setBool: NO forKey: @"ApplePressAndHoldEnabled"];
    }
}
```

This could be handled by GLFW by default, however I'm not sure if that would be desired or not?

Alternatively if you don't want to add any platform specific code, then you'll just need to avoid key callbacks. And instead manually poll the key states yourself.

**Comment by elmindreda** [comment]
This should be fixed now with 58a247b26d283c40078426443d375d4c126d69e1.

Thank you @laxcat for reporting it and @ScrimpyCat for pointing towards the solution!
